src/main/config/conversion/import/iso19139.ca.HNAP.FGP-to-iso19139.ca.HNAP.xsl

```diff
@@ -119,6 +119,32 @@
     </xsl:copy>
   </xsl:template>
 
+  <!--Add default LocalisedCharacterString to thumbnail -->
+  <xsl:template match="gmd:fileDescription">
```

Comment:
@wangf1122  - According to the following http://www.datypic.com/sc/niem21/e-gmd_fileDescription-1.html, there can be more elements than gco:CharacterString within gmd:fileDescription (although I have not seen any examples of this) - So I think you should use something like the following <xsl:apply-templates select="node()|@*"/> instead of specifically using gco:CharacterString

Also have you considered putting the condition in the template so that it only gets applied where needed? It is odd to have a template that starts with and if statement as that condition could have been placed in the template.

Something like the following (I have not tested)

<xsl:template match="gmd:fileDescription[gco:CharacterString/text() != ''  and not(gmd:PT_FreeText)]">
     <xsl:apply-templates select="node()|@*"/>
     <gmd:PT_FreeText>
          <gmd:textGroup>
            <xsl:choose>
              <xsl:when test="$mainLanguage='eng'">
                <gmd:LocalisedCharacterString locale="#fra">
                  <xsl:value-of select="./gco:CharacterString/text()"/>
                </gmd:LocalisedCharacterString>
              </xsl:when>
              <xsl:otherwise>
                <gmd:LocalisedCharacterString locale="#eng">
                  <xsl:value-of select="./gco:CharacterString/text()"/>
                </gmd:LocalisedCharacterString>
              </xsl:otherwise>
            </xsl:choose>
          </gmd:textGroup>
        </gmd:PT_FreeText>
</xsl:template>